bundles/framework/search/view/DefaultSearchView.js

```diff
@@ -83,11 +83,21 @@ Oskari.clazz.define(
             var button = this.getButton();
 
             var doSearch = function () {
+                field.setEnabled(false);
+                button.setEnabled(false);
             	me.__doSearch();
             };
+            var doAutocomplete = function() {
+                me.autocomplete = true;
+                me.__doSearch();
```

Comment:
I think this could be done with direct service call instead of changing the request API.